<launch>
	<!-- vim: set ft=xml noet : -->
	<!-- udp://127.0.0.1:14551@14555 -->

	<!-- <arg name="pilot_filename" default="fullIndoor23.txt" /> -->
    <arg name="pilot_filename" default="testVelocity.txt" />
	<arg name="config_filename" default="gazebo_config.txt" />
	<!-- image view params -->
	<arg name="front_cam" default="false" />
	<arg name="down_cam" default="false" />
	<arg name="env_percept" default="false" />
	<arg name="safety_node" default="false" />
	
	
	<env name="ROSCONSOLE_CONFIG_FILE" value="$(find uav_commander)/config/rosconsole.config" />
	<env name="ROSCONSOLE_FORMAT" value="[${severity}] [${time:%H:%M:%S}]: ${message}" />

	
  <node if="$(arg front_cam)" name="front_cam_image" pkg="image_view" type="image_view" output="screen">
    <remap unless="$(arg env_percept)" from="image" to="/logitech/webcam/webcam_image_raw" />
    <remap if="$(arg env_percept)" from="image" to="/image_percept/front_image/output_video" />
  </node>
  <node if="$(arg down_cam)" name="down_cam_image" pkg="image_view" type="image_view" output="screen">
    <remap unless="$(arg env_percept)" from="image" to="/logitech/fisheye/fisheye_image_raw" />
    <remap if="$(arg env_percept)" from="image" to="/image_percept/down_image/output_video" />
  </node>
  
  
  <node if="$(arg env_percept)" pkg="env_percept" type="img_percept_node" name="img_percept" output="screen" />

  <node name="pilot_node" pkg="uav_commander" type="pilot_node" args="$(find uav_commander)/pilot_files/$(arg pilot_filename) $(find uav_commander)/config/$(arg config_filename)" output="screen" respawn="true"/>
  
  <node if="$(arg safety_node)" name="safety_node" pkg="uav_commander" type="safety_node" output="screen" />

	<node pkg="uav_teleop" type="keyboard" name="keyboard" respawn="true" output="screen" />

</launch>
